1289/en.md

```diff
@@ -0,0 +1,145 @@
+# LightOJ 1289 - LCM from 1 to n
+
+*You are given `T` test cases, for each test case you are given `n`.*
+
+*You have to find `lcm(1, 2,...,n)` where lcm refers to the least common multiple of some integers, i.e. the smallest integer that is a multiple of all of the given integers (modulo 2^32).*
+
+---
+
+## Summary
+
+*lcm(b<sub>1</sub>, b<sub>2</sub>, ... , b<sub>m</sub>)* can be represented as *p<sub>1</sub><sup>a<sub>1</sub></sup> . p<sub>2</sub><sup>a<sub>2</sub></sup> . ... . p<sub>k</sub><sup>a<sub>k</sub></sup>* where *p<sub>1</sub>, p<sub>2</sub>, ... , p<sub>k</sub>* are prime factors of those numbers and *a<sub>1</sub>, a<sub>2</sub>, ... , a<sub>k</sub>* are the maximum powers of those corresponding primes, that occur in the numbers *b<sub>1</sub>, b<sub>2</sub>, ... , b<sub>k</sub>* . This **number theory** problem requires you to find the prime factors and their corresponding powers and the product of these.
+
+In this problem, you can assume *b<sub>1</sub>, b<sub>2</sub>, ... , b<sub>m</sub> = 1,2, ... , n* .
+
+## Solution
+
+### Naive Approach
+Let's calculate for each prime (until *n*) the maximum power *a* of this prime *p*, so that *p<sup>a</sup> <= n*. For example, if *n = 100*, prime *5* has highest power of *log<sub>5</sub>100*, i.e. the maximum number of time we can divide *100* by *5*, which is *2*. You can see, *5<sup>2</sup> = 25* which is less than *100*, and *5<sup>3</sup> = 125* which is greater than *100*. So here, for *n = 100*, *p = 5* we get *a = 2*.
+
+So, for each prime *p<sub>i</sub>* until *n*, we calculate *a<sub>i</sub>* , which is the maximum power of that prme till *n*. Then, *lcm =  p<sub>1</sub><sup>a<sub>1</sub></sup> . p<sub>2</sub><sup>a<sub>2</sub></sup> . ... . p<sub>k</sub><sup>a<sub>k</sub></sup>* , which is the required answer.
+
+#### Problem with Naive Solution
+*n* can be equal to *10<sup>8*, and *T* can be equal to *10<sup>4</sup>* . There are rougly *k = 6x10<sup>6</sup>* primes till  *10<sup>8* . So for each case, if we intend to use the naive solution, the complexity for each case would be *O(k log n)*, where *k* is the number of prime factors till *n*. So, total complexity will be *O(Tk log n)* , which won't pass the time limit of  4s.
```

Comment:
There is a minor typo. Please change `rougly` to `roughly`.

Reply:
I fixed the typo, thanks for letting me know!